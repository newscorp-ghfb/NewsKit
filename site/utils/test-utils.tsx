/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
