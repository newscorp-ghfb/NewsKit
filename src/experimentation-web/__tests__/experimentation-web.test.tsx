import * as React from 'react';
import {Helmet} from 'react-helmet';
import {ExperimentationWeb} from '..';
import {renderToFragment} from '../../test/test-utils';

jest.mock('../../utils/render-scripts');

describe('ExperimentationWeb', () => {
  [undefined, Helmet].forEach(helmet => {
    test('renders optimizely scripts', () => {
      const fragment = renderToFragment(
        <ExperimentationWeb
          optimizelyWebConfig={{
            scriptCdn: 'https://cdn.optimizely.com/js/123456789.js',
          }}
          reactHelmet={helmet}
        />,
      );
      expect(fragment).toMatchSnapshot();
    });

    test('renders optimizely scripts async', () => {
      const fragment = renderToFragment(
        <ExperimentationWeb
          optimizelyWebConfig={{
            scriptCdn: 'https://cdn.optimizely.com/js/123456789.js',
          }}
          reactHelmet={helmet}
          async
        />,
      );
      expect(fragment).toMatchSnapshot();
    });
  });
});
