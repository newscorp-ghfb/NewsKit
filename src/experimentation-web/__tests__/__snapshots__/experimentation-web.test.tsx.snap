// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentationWeb renders optimizely scripts 1`] = `
<DocumentFragment>
  <div>
    RenderScripts Component with props:{
  "scripts": [
    {
      "content": "(function() {\\n            if (window.location === window.parent.location) {\\n              window.optimizely_feature_flag = true;\\n              window[\\"optimizely_cdn\\"] = \\"https://cdn.optimizely.com/js/123456789.js\\";\\n\\n              window['optimizely'] = window['optimizely'] || [];\\n              if (document.cookie.indexOf('nuk-consent-personalisation=1') === -1) {\\n                window['optimizely'].push({\\n                  type: 'disable',\\n                });\\n              }\\n            }\\n          })();"
    },
    {
      "src": "https://cdn.optimizely.com/js/123456789.js",
      "async": false
    }
  ],
  "reactHelmet": "undefined"
}
  </div>
</DocumentFragment>
`;

exports[`ExperimentationWeb renders optimizely scripts 2`] = `
<DocumentFragment>
  <div>
    RenderScripts Component with props:{
  "scripts": [
    {
      "content": "(function() {\\n            if (window.location === window.parent.location) {\\n              window.optimizely_feature_flag = true;\\n              window[\\"optimizely_cdn\\"] = \\"https://cdn.optimizely.com/js/123456789.js\\";\\n\\n              window['optimizely'] = window['optimizely'] || [];\\n              if (document.cookie.indexOf('nuk-consent-personalisation=1') === -1) {\\n                window['optimizely'].push({\\n                  type: 'disable',\\n                });\\n              }\\n            }\\n          })();"
    },
    {
      "src": "https://cdn.optimizely.com/js/123456789.js",
      "async": false
    }
  ],
  "reactHelmet": "function"
}
  </div>
</DocumentFragment>
`;

exports[`ExperimentationWeb renders optimizely scripts async 1`] = `
<DocumentFragment>
  <div>
    RenderScripts Component with props:{
  "scripts": [
    {
      "content": "(function() {\\n            if (window.location === window.parent.location) {\\n              window.optimizely_feature_flag = true;\\n              window[\\"optimizely_cdn\\"] = \\"https://cdn.optimizely.com/js/123456789.js\\";\\n\\n              window['optimizely'] = window['optimizely'] || [];\\n              if (document.cookie.indexOf('nuk-consent-personalisation=1') === -1) {\\n                window['optimizely'].push({\\n                  type: 'disable',\\n                });\\n              }\\n            }\\n          })();"
    },
    {
      "src": "https://cdn.optimizely.com/js/123456789.js",
      "async": true
    }
  ],
  "reactHelmet": "undefined"
}
  </div>
</DocumentFragment>
`;

exports[`ExperimentationWeb renders optimizely scripts async 2`] = `
<DocumentFragment>
  <div>
    RenderScripts Component with props:{
  "scripts": [
    {
      "content": "(function() {\\n            if (window.location === window.parent.location) {\\n              window.optimizely_feature_flag = true;\\n              window[\\"optimizely_cdn\\"] = \\"https://cdn.optimizely.com/js/123456789.js\\";\\n\\n              window['optimizely'] = window['optimizely'] || [];\\n              if (document.cookie.indexOf('nuk-consent-personalisation=1') === -1) {\\n                window['optimizely'].push({\\n                  type: 'disable',\\n                });\\n              }\\n            }\\n          })();"
    },
    {
      "src": "https://cdn.optimizely.com/js/123456789.js",
      "async": true
    }
  ],
  "reactHelmet": "function"
}
  </div>
</DocumentFragment>
`;
